core/isolate.rs

```diff
@@ -748,8 +763,34 @@ pub mod tests {
     )
   }
 
+  struct DelayedFuture {
```

Comment:
Move this to its own file?

Reply:
Add docs too.

Reply:
It's used only in tests for isolate, seems superficial

Reply:
Oh this is only used in the tests. Nevermind about moving it to its own file..